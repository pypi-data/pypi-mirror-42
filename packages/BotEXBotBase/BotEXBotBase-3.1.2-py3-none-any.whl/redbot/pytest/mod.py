import pytest

__all__ = ["mod"]


@pytest.fixture
def mod(config, monkeypatch):
    from redbot.core import Config

    with monkeypatch.context() as m:
        m.setattr(Config, "get_conf", lambda *args, **kwargs: config)
        from redbot.core import modlog

        modlog._register_defaults()
        return modlog
