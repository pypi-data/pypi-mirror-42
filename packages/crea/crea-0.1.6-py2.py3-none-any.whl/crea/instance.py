# This Python file uses the following encoding: utf-8
from __future__ import absolute_import
from __future__ import division
from __future__ import print_function
from __future__ import unicode_literals
from builtins import object
import crea as stm


class SharedInstance(object):
    """Singelton for the Crea Instance"""
    instance = None
    config = {}


def shared_crea_instance():
    """ This method will initialize ``SharedInstance.instance`` and return it.
        The purpose of this method is to have offer single default
        crea instance that can be reused by multiple classes.

        .. code-block:: python

            from crea.account import Account
            from crea.instance import shared_crea_instance

            account = Account("test")
            # is equivalent with
            account = Account("test", crea_instance=shared_crea_instance())

    """
    if not SharedInstance.instance:
        clear_cache()
        SharedInstance.instance = stm.Crea(**SharedInstance.config)
    return SharedInstance.instance


def set_shared_crea_instance(crea_instance):
    """ This method allows us to override default crea instance for all users of
        ``SharedInstance.instance``.

        :param Crea crea_instance: Crea instance
    """
    clear_cache()
    SharedInstance.instance = crea_instance


def clear_cache():
    """ Clear Caches
    """
    from .blockchainobject import BlockchainObject
    BlockchainObject.clear_cache()


def set_shared_config(config):
    """ This allows to set a config that will be used when calling
        ``shared_crea_instance`` and allows to define the configuration
        without requiring to actually create an instance
    """
    if not isinstance(config, dict):
        raise AssertionError()
    SharedInstance.config.update(config)
    # if one is already set, delete
    if SharedInstance.instance:
        clear_cache()
        SharedInstance.instance = None
