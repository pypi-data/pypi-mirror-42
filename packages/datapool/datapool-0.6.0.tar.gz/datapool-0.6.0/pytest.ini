[pytest]
addopts = -rs --full-trace
testpaths = tests
