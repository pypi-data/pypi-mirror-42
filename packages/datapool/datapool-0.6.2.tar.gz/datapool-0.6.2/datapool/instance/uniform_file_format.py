# encoding: utf-8
from __future__ import print_function, division, absolute_import

import csv
import datetime
import io
import os

from datapool.utils import iter_to_list


def _check_header(header):

    header = list(header)

    if "site" in header:
        header.remove("site")
    else:
        for n in "xyz":
            if n not in header:
                raise ValueError(
                    "header {!r} must have either 'site' or 'x', 'y' and 'z'".format(
                        header
                    )
                )
            header.remove(n)

    header = set(header)
    fix_column_names = set(("timestamp", "parameter", "value"))
    if header != fix_column_names:
        missing = ["'{}'".format(name) for name in fix_column_names - header]
        invalid = ["'{}'".format(name) for name in header - fix_column_names]
        msg = ""
        if missing:
            msg += "missing: {}".format(", ".join(sorted(missing)))
        if invalid:
            msg += "invalid: {}".format(", ".join(sorted(invalid)))
        raise ValueError("invalid header ({})".format(msg))


@iter_to_list
def read_from_file(path):
    if not os.path.exists(path):
        raise ValueError("{} does not exist".format(path))

    with open(path, "r", encoding="ascii") as fh:
        yield from _read_from_fh(fh)


@iter_to_list
def read_from_string(data):
    fh = io.StringIO(data)
    yield from _read_from_fh(fh)


def can_be_converted_to_float(value):
    try:
        value = float(value)
    except ValueError:
        return False
    return True


def parse_timestamp(timestamp):
    return datetime.datetime.strptime(timestamp, "%Y-%m-%d %H:%M:%S")


def _read_from_fh(fh):
    reader = csv.reader(fh, delimiter=";")
    header = [f.strip() for f in next(reader)]
    _check_header(header)
    for i, row in enumerate(reader):
        if len(row) < len(header):
            # i + 1, we skipped header
            raise ValueError("row {} ({!r}) is incomplete".format(i + 1, row))
        if len(row) > len(header):
            # i + 1, we skipped header
            raise ValueError("row {} ({!r}) is to long".format(i + 1, row))

        row = (cell.strip() for cell in row)
        row_dict = dict(zip(header, row))
        yield row_dict


@iter_to_list
def check_rows(row_dicts):

    if not row_dicts:
        return

    for i, row_dict in enumerate(row_dicts):

        value = row_dict["value"]
        if not can_be_converted_to_float(value):
            yield "row {}: value '{}' is not a float".format(i, value)
        else:
            row_dict["value"] = float(value)

        timestamp = row_dict["timestamp"]
        try:
            row_dict["timestamp"] = parse_timestamp(timestamp)
        except ValueError:
            yield "row {}: timstamp '{}' either has wrong format or is invalid".format(
                i, timestamp
            )

        has_xyz = all(name in row_dict and row_dict[name] is not None for name in "xyz")
        not_empty = ("parameter",)
        if not has_xyz:
            not_empty += ("site",)

        for name in not_empty:
            if row_dict[name].strip() == "":
                yield "row {}: field {} is empty".format(i, name)


@iter_to_list
def to_signals(row_dicts):

    for row in row_dicts:
        for f in "xyz":
            if f in row:
                row["coord_" + f] = row[f]
                del row[f]

    from .domain_objects import Signal

    return list(map(Signal, row_dicts))
