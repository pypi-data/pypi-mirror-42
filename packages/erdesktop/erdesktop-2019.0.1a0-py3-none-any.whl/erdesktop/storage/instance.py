import os
import peewee

from erdesktop.settings import APP_DB_PATH, APP_DB_FILE


if not os.path.exists(APP_DB_PATH):
	os.makedirs(APP_DB_PATH)

DATABASE_INSTANCE = peewee.SqliteDatabase(APP_DB_FILE)
