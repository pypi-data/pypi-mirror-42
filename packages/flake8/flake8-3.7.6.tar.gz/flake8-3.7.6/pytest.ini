[pytest]
norecursedirs = .git .* *.egg* old docs dist build
addopts = -rw
filterwarnings =
    error
    # python3.4 raises this when importing setuptools
    ignore:The value of convert_charrefs will become True in 3.5.*:DeprecationWarning
