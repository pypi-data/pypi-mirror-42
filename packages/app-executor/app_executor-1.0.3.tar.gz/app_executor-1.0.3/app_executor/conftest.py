import pytest
import app_executor


@pytest.yield_fixture()
def executor(tmpdir):
    """
    return app_executor.AppExecutor object that will allow to run arbitrary number of
    processes. In teardown it will stop them and perform dump analysis. All results will be
    put into temporary directory.
    """
    with app_executor.AppExecutor(tmpdir) as app_exec:
        yield app_exec
