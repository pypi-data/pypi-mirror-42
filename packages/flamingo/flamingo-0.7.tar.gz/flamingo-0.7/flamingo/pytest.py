import pytest


@pytest.fixture
def dummy_context():
    from flamingo.core.data_model import ContentSet
    from flamingo.core.settings import Settings
    from flamingo.core.context import Context

    class DummyContext(Context):
        def __init__(self, settings, contents=None):
            self.settings = settings
            self.contents = contents or ContentSet()

    return DummyContext(settings=Settings())
