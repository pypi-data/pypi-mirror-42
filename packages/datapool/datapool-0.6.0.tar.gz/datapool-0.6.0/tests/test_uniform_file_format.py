# encoding: utf-8
from __future__ import absolute_import, division, print_function

import shutil

import pytest

from datapool.instance.uniform_file_format import (
    check_rows,
    read_from_file,
    read_from_string,
    to_signals,
)

from .conftest import not_for_root


def test_reader(regtest, data_path):
    rows = read_from_file(data_path("uniform_file.csv"))
    assert len(rows) == 2
    errors = check_rows(rows)
    assert not errors
    signals = to_signals(rows)

    for signal in signals:
        print(sorted(signal.items()), file=regtest)

    data = open(data_path("uniform_file.csv"), "r", encoding="ascii").read()
    rows = read_from_string(data)
    errors = check_rows(rows)
    assert not errors
    signals2 = to_signals(rows)

    assert signals == signals2

    with pytest.raises(ValueError):
        read_from_file(data_path("non_existent_uniform_file.csv"))


@not_for_root
def test_reader_wrong_permissions(tmpdir, data_path):
    new_path = tmpdir.join("file.csv")
    shutil.copyfile(data_path("uniform_file.csv"), new_path.strpath)
    new_path.chmod(0o000)

    with pytest.raises(IOError):
        read_from_file(data_path(new_path.strpath))


def test_wrong_header(data_path):
    with pytest.raises(ValueError):
        read_from_file(data_path("invalid_header_uniform_file.csv"))

    with pytest.raises(ValueError):
        read_from_file(data_path("invalid_header_uniform_file2.csv"))


def test_invalid_data_with_site(regtest, data_path):
    rows = read_from_file(data_path("invalid_data_uniform_file_site.csv"))
    for msg in check_rows(rows):
        print(msg, file=regtest)


def test_invalid_data_with_xyz(regtest, data_path):
    rows = read_from_file(data_path("invalid_data_uniform_file_xyz.csv"))
    for msg in check_rows(rows):
        print(msg, file=regtest)
