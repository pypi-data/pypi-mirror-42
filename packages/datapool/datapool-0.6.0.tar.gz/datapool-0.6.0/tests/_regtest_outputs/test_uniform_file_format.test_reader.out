[('coord_x', '10'), ('coord_y', '10'), ('coord_z', ''), ('parameter', 'water level'), ('site', None), ('timestamp', datetime.datetime(2016, 1, 1, 12, 13)), ('value', 123.45)]
[('coord_x', '10'), ('coord_y', '20'), ('coord_z', ''), ('parameter', 'water level'), ('site', None), ('timestamp', datetime.datetime(2016, 1, 2, 23, 0, 59)), ('value', 0.0)]
